qiskit/circuit/quantumcircuit.py

```diff
@@ -367,6 +342,24 @@ def layout(self) -> Optional[TranspileLayout]:
         """
         return self._layout
 
+    @classmethod
+    @property
+    @deprecate_func(
+        since="0.45.0", additional_msg="No alternative will be provided.", is_property=True
```

Comment:
I would add something like `handled by .qasm2.dumps, so no alternative is needed.`

Reply:
I dunno - the use of this object is/was already done by internally `QuantumCircuit.qasm`, so if someone's using it outside that context, then they presumably also don't want to use the new `qasm2.dumps`.